gamestonk_terminal/helper_funcs.py

```diff
@@ -22,6 +22,23 @@
     matplotlib.use(cfgPlot.BACKEND)
 
 
+def check_int_range(mini, maxi):
+    """
+    Checks if argparse argument is an int between 2 values.
+    https://stackoverflow.com/questions/55324449/how-to-specify-a-minimum-or-maximum-float-value-with-argparse
+    """
+
+    # Define the function with default arguments
+    def int_range_checker(num) -> int:
+        num = int(num)
+        if num < mini or num > maxi:
+            raise argparse.ArgumentTypeError(f"must be in range [{mini},{maxi}]")
+        return num
+
+    # Return function handle to checking function
+    return int_range_checker
```

Comment:
seems a bit redundant, it could have been just:
```
num = int(num)
if num < mini or num > maxi:
   raise argparse.ArgumentTypeError(f"must be in range [{mini},{maxi}]")
return num
```

Reply:
Thats what I initially thought, but the type has to callable, and doing it the way you suggest would never pass in a "num", no?

Reply:
Hmmm, I think I got it now. Never had seen code like this, but I think it makes sense. Really smart.

Basically you would do this only:
int_range_checker(num) -> int

but since you want to specify the min,max you declare that function a layer above so that you can define these thresholds on each different context.

Reply:
yeah.  Took me a while on stack exchange to find it lol.